src/concrete_intersection.jl

```diff
@@ -656,3 +656,27 @@ end
 function intersection(L1::LinearMap{N}, L2::LinearMap{N}) where {N}
     return intersection(linear_map(L1.M, L1.X), linear_map(L2.M, L2.X))
 end
+
+"""
+intersection(ca1::CartesianProductArray{N, S}, L2::CartesianProductArray{N, S}, nonzero_blocks::Vector{Int})
+              ) where {N<:Real, S<:LazySet{N}}
+
+Return the intersection of two CartesianProductArray's only for necessary blocks.
```

Comment:
"necessary" is not self-explanatory.
Also, since this argument should be optional (see my other comment), maybe better do not talk about this in the short description but add a `### Notes` section below.